frontend/packages/core/src/Resolver/hydrator.tsx

```diff
@@ -108,6 +108,11 @@ const OptionField = (
     });
   }, []);
 
+  const hasError = field.metadata.required && options.length === 0;
```

Comment:
if len === 0 and not required we should just omit the display of the dropdown altogether.